R/extract.R

```diff
@@ -1,5152 +1,5255 @@
-setMethod("extract", signature = className("mer", "lme4"),
-    definition = extract.mer)
-
-extract.lmerMod <- extract.lme4
-setMethod("extract", signature = className("lmerMod", "lme4"),
-    definition = extract.lmerMod)
-
-extract.glmerMod <- extract.lme4
-setMethod("extract", signature = className("glmerMod", "lme4"),
-    definition = extract.glmerMod)
-
-extract.nlmerMod <- extract.lme4
-setMethod("extract", signature = className("nlmerMod", "lme4"),
-    definition = extract.nlmerMod)
-
-
-# extension for lmrob and glmrob objects (robustbase package)
-extract.lmrob <- function(model, include.nobs = TRUE, ...) {
-  s <- summary(model, ...)
-
-  names <- rownames(s$coef)
-  co <- s$coef[, 1]
-  se <- s$coef[, 2]
-  pval <- s$coef[, 4]
-
-  gof <- numeric()
-  gof.names <- character()
-  gof.decimal <- logical()
-
-  if (include.nobs == TRUE) {
-    n <- length(model$residuals)
-    gof <- c(gof, n)
-    gof.names <- c(gof.names, "Num.\ obs.")
-    gof.decimal <- c(gof.decimal, FALSE)
-  }
-
-  tr <- createTexreg(
-    coef.names = names,
-    coef = co,
-    se = se,
-    pvalues = pval,
-    gof.names = gof.names,
-    gof = gof,
-    gof.decimal = gof.decimal
-  )
-  return(tr)
-}
-
-setMethod("extract", signature = className("lmrob", "robustbase"),
-    definition = extract.lmrob)
-
-extract.glmrob <- extract.lmrob
-setMethod("extract", signature = className("glmrob", "robustbase"),
-    definition = extract.glmrob)
-
-
-# extension for lnam objects (sna package)
-extract.lnam <- function(model, include.rsquared = TRUE, include.adjrs = TRUE,
-    include.aic = TRUE, include.bic = TRUE, include.loglik = TRUE, ...) {
-  coefs <- coef(model, ...)
-  coef.names <- names(coefs)
-  se <- c(model$beta.se, model$rho1.se, model$rho2.se)
-  p <- 2 * (1 - pnorm(abs(coefs), 0, se))
-
-  rss <- sum(model$residuals^2)
-  mss <- sum((model$fitted - mean(model$fitted))^2)
-  rdfns <- model$df.residual + 1
-  rsquared <- mss / (mss + rss)
-  adj.rsquared <- 1 - (1 - mss / (mss + rss)) * model$df.total / rdfns
-  lik <- model$lnlik.model
-  aic <- -2 * model$lnlik.model + 2 * model$df.model
-  bic <- -2 * model$lnlik.model + log(model$df.total) * model$df.model
-
-  gof <- numeric()
-  gof.names <- character()
-  gof.decimal <- logical()
-  if (include.rsquared == TRUE) {
-    gof <- c(gof, rsquared)
-    gof.names <- c(gof.names, "R$^2$")
-    gof.decimal <- c(gof.decimal, TRUE)
-  }
-  if (include.adjrs == TRUE) {
-    gof <- c(gof, adj.rsquared)
-    gof.names <- c(gof.names, "Adj.\ R$^2$")
-    gof.decimal <- c(gof.decimal, TRUE)
-  }
-  if (include.aic == TRUE) {
-    gof <- c(gof, aic)
-    gof.names <- c(gof.names, "AIC")
-    gof.decimal <- c(gof.decimal, TRUE)
-  }
-  if (include.bic == TRUE) {
-    gof <- c(gof, bic)
-    gof.names <- c(gof.names, "BIC")
-    gof.decimal <- c(gof.decimal, TRUE)
-  }
-  if (include.loglik == TRUE) {
-    gof <- c(gof, lik)
-    gof.names <- c(gof.names, "Log Likelihood")
-    gof.decimal
```

Comment:
It strikes me that this function looks very much like many other functions. Can you check if there is an identical function somewhere in the file? If that's the case, we can just create a copy and can avoid all the duplication...